packages/x-data-grid/src/components/cell/GridBooleanCell.tsx

```diff
@@ -140,7 +140,7 @@ const GridBooleanCell = React.memo(GridBooleanCellRaw);
 export { GridBooleanCell };
 
 export const renderBooleanCell: GridColDef['renderCell'] = (params: GridBooleanCellProps) => {
-  if (isAutoGeneratedRow(params.rowNode)) {
+  if (params.field !== '__row_group_by_columns_group__' && isAutoGeneratedRow(params.rowNode)) {
```

Comment:
Updated logic doesn't show any content if row is `autogenerated` and row is not grouped by that column.